apps/api/src/domain/models/digital-text/commands/create-digital-text.command-handler.ts

```diff
@@ -0,0 +1,146 @@
+import { CommandHandler } from '@coscrad/commands';
+import { Inject } from '@nestjs/common';
+import { InternalError, isInternalError } from '../../../../lib/errors/InternalError';
+import { ValidationResult } from '../../../../lib/errors/types/ValidationResult';
+import { isNotAvailable } from '../../../../lib/types/not-available';
+import { isNotFound } from '../../../../lib/types/not-found';
+import { isOK } from '../../../../lib/types/ok';
+import { REPOSITORY_PROVIDER_TOKEN } from '../../../../persistence/constants/persistenceConstants';
+import { DTO } from '../../../../types/DTO';
+import { ResultOrError } from '../../../../types/ResultOrError';
+import { buildMultilingualTextWithSingleItem } from '../../../common/build-multilingual-text-with-single-item';
+import { Valid } from '../../../domainModelValidators/Valid';
+import getInstanceFactoryForResource from '../../../factories/getInstanceFactoryForResource';
+import { EVENT, IIdManager } from '../../../interfaces/id-manager.interface';
+import { IRepositoryForAggregate } from '../../../repositories/interfaces/repository-for-aggregate.interface';
+import { IRepositoryProvider } from '../../../repositories/interfaces/repository-provider.interface';
+import { AggregateId } from '../../../types/AggregateId';
+import { InMemorySnapshot, ResourceType } from '../../../types/ResourceType';
+import buildInMemorySnapshot from '../../../utilities/buildInMemorySnapshot';
+import { BaseCommandHandler } from '../../shared/command-handlers/base-command-handler';
+import UuidNotGeneratedInternallyError from '../../shared/common-command-errors/UuidNotGeneratedInternallyError';
+import { BaseEvent } from '../../shared/events/base-event.entity';
+import { DigitalText } from '../digital-text.entity';
+import { CreateDigitalText } from './create-digital-text.command';
+import { DigitalTextCreated } from './digital-text-created.event';
+
+@CommandHandler(CreateDigitalText)
+export class CreateDigitalTextCommandHandler extends BaseCommandHandler<DigitalText> {
```

Comment:
I was working from the `create-song.command-handler.ts` which just extends `BaseCommandHandler`.  Is the `BaseCreateCommandHandler` still ok to use for event sourcing?

Reply:
It's got nothing to do with event sourcing. It has to do with the fact that `BaseCreateCommandHandler` did not exist when we wrote `create-song.command-handler.ts` and we didn't want to add it until the "pattern emerged", so not to over-engineer. 

Since the actual command executor knows nothing about the internals of the handlers, aside form the fact that they have an `async execute(fsa: CommandFSA): Error | Ack`, it doesn't actually break anything, so we didn't see much value in refactoring. Next time we work on the song model, we can address this. 

Main lesson to learn is to use git lens and see how old the file you are using for "inspiration" is, or to watch PRs for new commands more closely so you can keep up to date on the latest approaches \ best practices.